src/main/scala/com/databricks/spark/csv/CsvReader.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2015 Ayasdi Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ayasdi.bigdf.readers
+
+import java.io.StringReader
+import com.univocity.parsers.csv._
+
+abstract class CsvReader(fieldSep: Char = ',',
+                         lineSep: String = "\n",
+                         quote: Char = '"',
+                         escape: Char = '\\',
+                         ignoreLeadingSpace: Boolean = true,
+                         ignoreTrailingSpace: Boolean = true,
+                         headers: Seq[String],
+                         inputBufSize: Int = 128,
+                         maxCols: Int = 20480) {
```

Comment:
I'm not sure if the code that is brought in as an entire file should be modified, but this style of parameter indentation is different from what being used elsewhere in the code (e.g. java.io.Reader).